tsdb/shard.go

```diff
@@ -283,20 +528,18 @@ func (s *Shard) deleteMeasurement(name string, seriesKeys []string) error {
 			if err := b.Delete([]byte(k)); err != nil {
 				return err
 			}
-			if err := tx.DeleteBucket([]byte(k)); err != nil {
+			if err := tx.DeleteBucket([]byte(k)); err != nil && err != bolt.ErrBucketNotFound {
 				return err
 			}
+			delete(s.cache, k)
 		}
 
 		return nil
 	}); err != nil {
-		_ = s.Close()
```

Comment:
I think `disk full` or whole box issues should cause the process to panic. Better to go down hard and set off alarms than stay in some odd state where it's kind of working for somethings, but not for others.